src/jit/lowerxarch.cpp

```diff
@@ -2521,42 +2533,83 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* containingNode, Ge
     noway_assert(supportsRegOptional != nullptr);
     *supportsRegOptional = supportsGeneralLoads;
 
-    if (!node->OperIsHWIntrinsic())
+    if (!node->OperIsHWIntrinsic() && !(category == HW_Category_MemoryLoad && nodeOper == GT_LEA && supportsContainment))
```

Comment:
This temporary solution until whole category will have support for address folding